src/apiv2.ts

```diff
@@ -0,0 +1,272 @@
+import fetch, { Response, RequestInit } from "node-fetch";
+
+import { FirebaseError } from "./error";
+import * as logger from "./logger";
+import * as responseToError from "./responseToError";
+import * as scopes from "./scopes";
+
+const CLI_VERSION = require("../package.json").version;
+
+const VALID_METHODS = new Set(["GET", "PUT", "POST", "DELETE", "PATCH"]);
+
+type FirebaseRequestOptions = {
+  method?: string;
+  baseURL: string;
```

Comment:
How about just `urlPrefix` (assuming you don't want to keep `origin` because *technically* it's not always set to an origin?